### ISSUE TITLE
Don't hardcode path to ios project

### ISSUE THREAD

**Issue by stianjensen** [microsoft/appcenter-sdk-react-native#137]
In MobileCenterCrashes and MobileCenterAnalytics, the FRAMEWORK_SEARCH_PATHS key is hardcoded in the package's respective project files so that the react-native project's ios folder _has_ to be called `ios`:
https://github.com/Microsoft/mobile-center-sdk-react-native/blob/0feed4207aabd0bf2b14dd2932bd5a68b9f56f33/mobile-center-crashes/ios/RNCrashes.xcodeproj/project.pbxproj#L236-L246

Similar packages like Reacts RCTLinking project, does not do this:
https://github.com/facebook/react-native/blob/acd9a29d94f145ae5b1c755ace7bb71964e363c6/Libraries/LinkingIOS/RCTLinking.xcodeproj/project.pbxproj#L256-L260

Would it be possible to remove the hardcoded FRAMEWORK_SEARCH_PATHS to support projects where the main folder is not called `ios`? Specifically, I need this because I have two ios projects that are built in parallell from the same repository with the same dependencies.

**Comment by bmourat** [comment]
Hi @stianjensen ,

As I understand you correctly, for your project you need a mechanism to set framework search path manually from your app project? Or is it something else? Could please elaborate on your setup: are those two different targets in one project or two different projects?
 
Best,
Murat

**Comment by stianjensen** [comment]
Sure, here's my setup:
I make two different apps from the same react native code base (because they mainly use all the same components and views, and they're always kept in sync).
For each of the two apps there is an android app and and ios app, so essentially there are four native apps.

I've set this up as two separate XCode projects and two native android projects as follows:
```
ios-one/
    Podfile
    Podfile.lock
    ios-one/
    ios-one.xcodeproj
ios-two/
    Podfile
    Podfile.lock
    ios-two/
    ios-two.xcodeproj
android-one/
    ...
android-two/
    ...
src/
node_modules/
package.json
```

Essentially I'm wondering if it is possible to discover the native framework that you depend on without hardcoding the path to the main project folder so it has to be called `ios`. I have other native dependencies and hadn't run into this issue before integrating this library.
For instance, as I commented, I saw that RCTLinking had no need to specify the FRAMEWORK_SEARCH_PATHS. I'm not fully aware of how this all works though, so I get it if this project has needs that require it to be specified that RCTLinking does not have.

My current workaround is to add a symlink for the project I'm currently building, so that the path ../../../ios/ will temporarily resolve when I need it to, like this: `ln -s ios-one ios`.

**Comment by RyanThomas73** [comment]
@stianjensen I wanted to explain some more about `FRAMEWORK_SEARCH_PATHS`, differences between the appcenter react-native sdk and RCTLinking, and how you can work around this.

One of the big differences between `RCTLinking` and the `AppCenter-SDK-React-Native` is that the latter depends on additional frameworks that `RCTLinking` does not. To illustrate this you can look at the @import statements in (currently at line no. 19/20):
https://github.com/Microsoft/AppCenter-SDK-React-Native/blob/develop/appcenter/ios/AppCenterReactNative/AppCenterReactNative.m

`AppCenter-SDK-React-Native` depends on the underlying `AppCenter-SDK-Apple` to be installed in your ios project(s) and needs this framework search path to find it. Looking at `RCTLinking` I don't see any such framework imports in the `.h` / `.m` files so it wouldn't need the framework search paths.

I haven't tested this, however the AppCenter framework search paths include the `$(inherited)` path so you should be able to work around this by including your relative search path to your `Pods/` folder in your own project and it should be inherited. (e.g in ios-one's search paths you could include `$(SRCROOT)/Pods/**` and it should be inherited).


Needing to manually add this, however, is contrary to the AppCenter documentation which encourages using `react-native link` to handle everything. My PR includes updating these framework search paths to match the ios project path determined by the react-native config. Once my PR is merged you should be able to link each project correctly using a cli calls like this:
* `react-native link appcenter appcenter-analytics appcenter-crashes --config path/to/ios-ones/rn-cli.config.js`
* `react-native link appcenter appcenter-analytics appcenter-crashes --config path/to/ios-two/rn-cli.config.js`

Hope this helps.

**Comment by DmitriyKirakosyan** [comment]
I'm going to close this issue as a duplicate of https://github.com/microsoft/appcenter-sdk-react-native/issues/220.
Please reopen if you feel it is different.
